databases/week2/databases_week2.sql

```diff
@@ -0,0 +1,487 @@
+JOIN 
+task ON user_task.task_id = task.id
+WHERE email LIKE '%@spotify.com';
+
+-- Get all the tasks for 'Donald Duck' with status 'Not started'
+
+SELECT  task.title, status.name
+FROM user
+JOIN user_task ON user.id = user_task.user_id
+JOIN
+task ON task.id = user_task.task_id
+JOIN status ON task.status_id = status.id
+WHERE user.name like '%Donald Duck'
+AND status.name = 'Not started';
+
+-- Get all the tasks for 'Maryrose Meadows' that were created in september (hint: month(created)=month_number
+
+SELECT task.title, user.name , task.created
+FROM user
+JOIN user_task ON user_task.user_id= user.id
+JOIN
+task ON user_task.task_id = task.id
+WHERE user.name = 'Maryrose Meadows' AND MONTH(created) = 09;
+
+-- How many tasks where created in each month
+
+SELECT COUNT(*) ,MONTHNAME(created)
+FROM task GROUP BY MONTHNAME(created);
+
+
+
+
+
+
+-- Part 4: Creating a database
+
+
+
+-- MySQL Workbench Forward Engineering
+
+SET @OLD_UNIQUE_CHECKS=@@UNIQUE_CHECKS, UNIQUE_CHECKS=0;
+
+SET @OLD_FOREIGN_KEY_CHECKS=@@FOREIGN_KEY_CHECKS, FOREIGN_KEY_CHECKS=0;
+
+SET @OLD_SQL_MODE=@@SQL_MODE, SQL_MODE='ONLY_FULL_GROUP_BY,STRICT_TRANS_TABLES,NO_ZERO_IN_DATE,NO_ZERO_DATE,ERROR_FOR_DIVISION_BY_ZERO,NO_ENGINE_SUBSTITUTION';
+
+
+-- -------------------------------------------------------
+ Schema creating_database
+-------------------------------------------------------
+
+---------------------------------------------------------
+ Schema creating_database
+
+-------------------------------------------------------
+
+
+CREATE SCHEMA IF NOT EXISTS `creating_database` DEFAULT CHARACTER SET utf8 ;
+
+USE `creating_database` ;
+
+
+---------------------------------------------------------
+
+ Table `creating_database`.`students`
+
+------------------------------------------------------
+
+CREATE TABLE IF NOT EXISTS `creating_database`.`students` (
+
+
+`idstudents` INT NOT NULL AUTO_INCREMENT,
+ 
+`staff` VARCHAR(45) NOT NULL,
+
+`offices` VARCHAR(45) NOT NULL,
+
+`job_titles` VARCHAR(45) NULL,
+
+PRIMARY KEY (`idstudents`))
+
+ENGINE = InnoDB;
+
+
+
+---------------------------------------------------------
+
+Table `creating_database`.`kitchen`
+
+-------------------------------------------------------
+
+CREATE TABLE IF NOT EXISTS `creating_database`.`kitchen` (
+
+`idkitchen` INT NOT NULL AUTO_INCREMENT,
+
+`dishes` VARCHAR(45) NOT NULL,
+
+`drinks` VARCHAR(45) NOT NULL,
+
+`desert` VARCHAR(45) NOT NULL,
+
+`menu` INT NOT NULL,
+
+PRIMARY KEY (`idkitchen`))
+ENGINE = InnoDB;
+
+
+
+
+---------------------------------------------------------
+
+Table `creating_database`.`kitchen_has_students`
+
+-------------------------------------------------------
+
+
+CREATE TABLE IF NOT EXISTS `creating_database`.`kitchen_has_students` (
+ 
+`kitchen_idkitchen` INT NOT NULL,
+
+`students_idstudents` INT NOT NULL,
+ 
+PRIMARY KEY (`kitchen_idkitchen`, `students_idstudents`),
+ 
+INDEX `fk_kitchen_has_students_students1_idx` 
```

Comment:
I hope you understood the concept of INDEX and `fk_kitchen_has_students_students1_idx`.
What is the use case of VISIBLE?